### DISCUSSION TITLE
How do I change the default value of "rocm-device-lib-path"?

### DISCUSSION THREAD

**Discussion by biergaizi** [intel/llvm#9747] [Q&A]
My system installs ROCm to locations unknown to the DPC++ build system, so when building DPC++, I need to use `DSYCL_BUILD_PI_HIP_ROCM_DIR`, `DSYCL_BUILD_PI_HIP_INCLUDE_DIR`, `DSYCL_BUILD_PI_HIP_HSA_INCLUDE_DIR`, and `DSYCL_BUILD_PI_HIP_LIB_DIR`. I was able to complete the build.

However, after installing, targeting HIP ROCm creates the following error message:

    clang++: error: cannot find ROCm device library; provide its path via '--rocm-path' or
    '--rocm-device-lib-path', or pass '-nogpulib' to build without ROCm device library

I was able to solve the problem by passing `--rocm-device-lib-path=/usr/lib/amdgcn/bitcode/` manually. But is there a way to change this search path at build time? Which build-time variable is controlling it?

**Comment by jinz2014** [comment]
The DPC++ build assumes that ROCm is installed in /opt/rocm, if it is installed somewhere else, the directory must be provided through the CMake variable SYCL_BUILD_PI_HIP_ROCM_DIR which can be passed using the --cmake-opt option of configure.py as follows:

python $DPCPP_HOME/llvm/buildbot/configure.py --hip \
  --cmake-opt=-DSYCL_BUILD_PI_HIP_ROCM_DIR=/usr/local/rocm
